src/EditorFeatures/Core.Wpf/Lightup/LightupHelpers.cs

```diff
@@ -328,21 +329,26 @@ public static Action<T> CreateActionAccessor<T>(Type? type, string methodName)
             throw new InvalidOperationException($"Method '{method}' produces a value of type '{method.ReturnType}', which is not assignable to type '{typeof(TResult)}'");
         }
 
-        var parameter = Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
-        var argument = Expression.Parameter(typeof(TArg), parameters[0].Name);
-        var instance =
+        var instanceParameter =
             type.GetTypeInfo().IsAssignableFrom(typeof(T).GetTypeInfo())
-            ? (Expression)parameter
-            : Expression.Convert(parameter, type);
+                ? Expression.Parameter(type, GenerateParameterName(type))
+                : Expression.Parameter(typeof(T), GenerateParameterName(typeof(T)));
+
+        var argument = Expression.Parameter(typeof(TArg), parameters[0].Name);
         var convertedArgument =
             argType.GetTypeInfo().IsAssignableFrom(typeof(TArg).GetTypeInfo())
             ? (Expression)argument
             : Expression.Convert(argument, argType);
 
         var expression =
             Expression.Lambda<Func<T, TArg, TResult>>(
-                Expression.Convert(Expression.Call(instance, method, convertedArgument), typeof(TResult)),
-                parameter);
+                Expression.Convert(
```

Comment:
There is an exception in runtime in this expression, because it is constructing a lambda (Func<T, TArg, TResult>, two parameters), but only passing one parameter. I rewrote this part so it creates something like this 
```
TResult Expression<T, Arg, TResult>(T instanceParameter, Arg argument)
{
     var temp = ((type)instanceParameter).method(argument);
     return (TResult)temp; 
}
```

Reply:
cc @sharwell